operator/market-indexer/framework/migrations/20250919140741_init.sql

```
@@ -0,0 +1,31 @@
+-- Up
+
+-- Job events table
+CREATE TABLE job_events (
+    id BIGSERIAL PRIMARY KEY,
+    job_id VARCHAR(66) NOT NULL,
+    event_name VARCHAR(255) NOT NULL,
+    event_data JSONB NOT NULL,
+    indexer_process_time TIMESTAMPTZ DEFAULT now()
+);
+
+-- Indexer state table to track progress
+CREATE TABLE indexer_state (
+    id INT PRIMARY KEY,
+    last_processed_block BIGINT NOT NULL,
+    updated_at TIMESTAMPTZ DEFAULT now()
+);
+
+-- Initial values
+INSERT INTO indexer_state (id, last_processed_block) VALUES (1, -1);
+
+-- Useful index (for querying active job_id's based on the JobOpened & JobClosed events)
+CREATE INDEX idx_job_events_event_name_job_id ON job_events (event_name, job_id); 
```

the cp would also probably use some indexes right? if so, are we adding them here or in the cp PR

---

anyway, can add it in the cp PR based on the query there

---

random panics :man_shrugging:

---

tasks can panic for reasons out of our control, like from some library or allocation failures or whatever

---

> random panics 🤷‍♂️

In the original design the job manager task only restarted when the last `JobResult` status was `Retry` which only was the case for randomly ended events stream and nothing else

---

sure, that swallows panics as well I'm guessing

---

> > tasks can panic for reasons out of our control, like from some library or allocation failures or whatever
> 
> But since CP is such a crucial service the ideal behavior for any such random panic should be to abort and restart it because if it's a random reason then what is the guarantee it won't happen again and again

because initial sync takes time, or used to at least, idk with the current architecture

---

> sure, that swallows panics as well I'm guessing

I don't think so, Can you illustrate through an example?

---

sure, job task panics for whatever reason, task is killed by tokio, the job task now doesn't exist forever till a cp restart

---

> > But it was the case originally as well, also what if for that whatever reason the other tasks also starts getting killed? I think we should design a service in such a way that it covers most of the error and corner cases which we have done because the original CP works fine afaik and even then if the service panics then it's best to abort and investigate the issue
> 
> aborting on panics is an option, but only if initial sync is quick (< 10s)

Well that'll depend on the operator, how many jobs it was assigned and how many was terminated things like that, but looking at the current usage and jobs stats I think it should be doable

---

ditch for now, we'll discuss in the cp pr